sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -339,8 +339,11 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with Logging {
       case SqlBaseParser.SELECT =>
         // Regular select
 
+        // Add hints.
+        val withHint = relation.optionalMap(ctx.hint)(withHints)
```

Comment:
Yep. I grasp your point. First, the given `relation` is just a logical plan, not a table. In the following case, the `relation` is `Join`.
 ```
SELECT /*+ MAPJOIN(parquet_t0) */ * FROM parquet_t0, parquet_t1
```
This means just a context. 

Second, I agree with your point. Generally, `hint` is about tables like join order or access path. We can provides some hints on filter conditions, I'm not sure about the cases so far. But, even for that cases, we can adapt that in `Analyzer` easily. In other words, we can identify `Hint("FILTERHINT")` in the same SPJ.